templates/exchange.template

```diff
@@ -0,0 +1,5929 @@
+                            ]
+                        ]
+                    }
+                }
+            },
+            "Type": "AWS::EC2::Instance"
+        },
+        "WSFCNode1": {
+            "Metadata": {
+                "AWS::CloudFormation::Authentication": {
+                    "S3AccessCreds": {
+                        "type": "S3",
+                        "roleName": {
+                            "Ref": "ExchangeRole"
+                        },
+                        "buckets": [
+                            {
+                                "Ref": "QSS3BucketName"
+                            }
+                        ]
+                    }
+                },
+                "AWS::CloudFormation::Init": {
+                    "Cleanup": {
+                        "commands": {
+                            "a-disable-credssp": {
+                                "command": "powershell.exe -ExecutionPolicy RemoteSigned -Command \"C:\\cfn\\scripts\\DisableCredSsp.ps1\"",
+                                "waitAfterCompletion": "0"
+                            },
+                            "b-enable-uac": {
+                                "command": "powershell.exe -ExecutionPolicy RemoteSigned -Command \"C:\\cfn\\scripts\\Configure-UAC.ps1 -status Enable\"",
+                                "waitAfterCompletion": "0"
+                            }
+                        }
+                    },
+                    "Finalize": {
+                        "commands": {
+                            "a-reboot": {
+                                "command": "powershell.exe -Command \"Restart-Computer -Force\"",
+                                "waitAfterCompletion": "forever"
+                            },
+                            "b-signal-success": {
+                                "command": "powershell -Command \"Write-AWSQuickStartStatus\""
+                            }
+                        }
+                    },
+                    "Install": {
+                        "commands": {
+                            "a-install-ucma-runtime": {
+                                "command": "powershell.exe -Command \"C:\\cfn\\scripts\\Install-UcmaRuntime.ps1\"",
+                                "waitAfterCompletion": "0"
+                            },
+                            "b-install-exchange-organization": {
+                                "command": {
+                                    "Fn::Join": [
+                                        "",
+                                        [
+                                            "powershell.exe -Command \"C:\\cfn\\scripts\\Install-ExchangeOrg.ps1 -DomainNetBIOSName '",
+                                            {
+                                                "Ref": "DomainNetBIOSName"
+                                            },
+                                            "' -DomainAdminUs
```

Comment:
I need a bit more clarify around required ports and who communicated with who, as it seems that there is quite an overlap from WSFC allowing all ports but then ExchangeServerAccessSecurityGroup being used for select ports, then Edge server security group allowing a 0.0.0.0/0 on other ports. Can we discuss this a bit more?